core/modules/replica/ConfigurationStore.cc

```diff
@@ -262,6 +262,70 @@ WorkerInfo ConfigurationStore::setWorkerDataDir(string const& name,
 }
 
 
+WorkerInfo ConfigurationStore::setWorkerDbHost(std::string const& name,
+                                               std::string const& host) {
+    LOGS(_log, LOG_LVL_DEBUG, context(__func__) << "  name=" << name << " host=" << host);
+    util::Lock lock(_mtx, context(__func__));
+
+    auto&& itr = _workerInfo.find(name);
+    if (_workerInfo.end() == itr) {
+        throw invalid_argument(
+                ::classMethodContext(__func__) + "  no such worker: " + name);
+    }
+    itr->second.dbHost = host;
+
+    return itr->second;
+}
+
+
+WorkerInfo ConfigurationStore::setWorkerDbPort(std::string const& name,
+                                               uint16_t port) {
+    LOGS(_log, LOG_LVL_DEBUG, context(__func__) << "  name=" << name << " port=" << port);
+    util::Lock lock(_mtx, context(__func__));
+
+    auto&& itr = _workerInfo.find(name);
+    if (_workerInfo.end() == itr) {
+        throw invalid_argument(
+                ::classMethodContext(__func__) + "  no such worker: " + name);
+    }
```

Comment:
Since I don't like using C/C++ macros here then I may go for the following code pattern instead (which would be using a member function):
```
assertIfTrue(_workerInfo.end() == ptr, __func__, "  no such worker: " + name);
```

Reply:
Refactored. It turns out I already have a base class's method which encapsulates this pattern:  `Configuration::safeFindWorker(...)`. This particular class was not refactored to use that method.